### PR TITLE
CLOUDP-131680: [atlascli] atlas backups restores watch <restoreJobId> [flags]

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-131680

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/atlas/backup/restores/watch.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		return err
+	}
+}
+
+func (opts *WatchOpts) watcher() (bool, error) {
+	result, err := opts.store.RestoreJob(opts.ConfigProjectID(), opts.clusterName, opts.id)
+	if err != nil {
+		return false, err
+	}
+	return result.Expired || result.Cancelled || *result.Failed || result.FinishedAt != "", nil
```
Comment (andreaangiolillo) [inline internal/cli/atlas/backup/restores/watch.go:51]:
```suggestion
	return result.Expired || result.Cancelled || result.Failed || result.FinishedAt != "", nil
```

File: internal/cli/atlas/backup/restores/watch.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+func (opts *WatchOpts) Run() error {
+	if err := opts.Watch(opts.watcher); err != nil {
+		return err
+	}
+
+	return opts.Print(nil)
+}
+
+// mongocli atlas backup(s) restore(s) job(s) watch <ID>.
```
Comment (andreaangiolillo) [inline internal/cli/atlas/backup/restores/watch.go:62]:
```suggestion
// atlas backup(s) restore(s) job(s) watch <ID>.
```

File: internal/cli/atlas/backup/restores/watch.go (hunk 1 above)
Comment (gssbzn) [reply internal/cli/atlas/backup/restores/watch.go:51]: `result.Failed` is a pointer, I need to dereferenced it to compare with non pointers

Comment (andreaangiolillo) [reply internal/cli/atlas/backup/restores/watch.go:51]: Thanks for the clarification! : )

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: internal/cli/atlas/backup/restores/watch.go (hunk 2 above)
Comment (gssbzn) [reply internal/cli/atlas/backup/restores/watch.go:62]: I'll update if I have comments from docs

**Review by sarahsimpers — APPROVED** [review]
LGTM

**Pull request merged**